lib/internal/streams/pipeline.js

```diff
@@ -178,8 +180,41 @@ function pipeline(...streams) {
 
   let finishCount = 0;
 
+  let hasTransform = [];
+
   function finish(err) {
-    const final = --finishCount === 0;
+    hasTransform = streams.filter((s) => {
+      try {
+        const hasTransformContext =
+          s.constructor.toString().indexOf('Transform') > -1;
+        const hasNotPassThrough =
+          s.constructor.toString().indexOf('PassThrough') === -1;
+        const hasNotWritable =
+          s.constructor.toString().indexOf('Writable') === -1;
+        const hasNotAsyncGeneratorFunction =
+          s.constructor.toString().indexOf('AsyncGeneratorFunction') === -1;
+        const hasNotFlush =
+          s._flush === undefined;
+        return (
+          hasTransform.length === 0 &&
```

Comment:
Maybe it's just me but what's the point of this? It seems very confusing to check the length of the array that you're assigning to... 😬